packages/core-modules/contracts/modules/CoreElectionModule.sol

```diff
@@ -176,8 +174,15 @@ contract CoreElectionModule is IElectionModule, ElectionStorage, OwnableMixin {
         }
 
         if (electionData.addressVoted[msg.sender]) {
-            // TODO: if msg.sender already voted, rollback previous votes and allow to re-vote
-            revert AlreadyVoted();
+            uint voteDataIndex = electionData.votesIndexes[msg.sender];
+            VoteData memory previousVoteData = electionData.votes[voteDataIndex];
+
+            for (uint i = 0; i < previousVoteData.candidates.length; i++) {
+                electionData.nomineeVotes[previousVoteData.candidates[i]] += previousVoteData.votePower;
+            }
+
+            // Cleanup previous vote in store
+            electionData.votes[voteDataIndex] = VoteData({candidates: new address[](0), votePower: 0});
```

Comment:
Hmm, not 100% about this, but it doesn't initialize the values by itself? like so:

```suggestion
            electionData.votes[voteDataIndex] = VoteData();
```

Reply:
Yeah, thought the same thing. Could work.

Reply:
Nop. It's not allowed by the compiler. It failed with an error saying there are missing arguments. That was my first, natural choice.